packages/stats/src/index.ts

```diff
@@ -0,0 +1,159 @@
+import * as Parser from '@tracespace/parser'
+
+export interface DrillUsage {
+  diameter: number
+  count: number
+}
+
+export interface DrillStats {
+  drillHits: DrillUsage[]
+  drillRoutes: DrillUsage[]
+  totalDrills: number
+  totalRoutes: number
+  minDrillSize: number
+  maxDrillSize: number
+}
+
+/**
+ * Method to collect drill stats of parsed drill files.
+ *
+ * @example
+ * ```ts
+ * import {createParser} from '@tracespace/parser'
+ * import {collectDrillStats} from '@tracespace/stats'
+ *
+ * const parser = createParser()
+ *
+ * parser.feed(...)
+ *
+ * const tree = parser.results()
+ * const stats = collectDrillStats([tree])
+ * ```
+ *
+ * @category Stats
+ */
+export function collectDrillStats(trees: Parser.Root[]): DrillStats {
+  const state: DrillStatsState = {
+    usedTools: {},
+    drillsPerTool: {},
+    routesPerTool: {},
+    minDrillSize: null,
+    maxDrillSize: null,
+  }
+
+  for (const tree of trees) {
+    if (tree.filetype !== Parser.DRILL) {
+      continue
+    }
+
+    _updateDrillStats(state, tree)
+  }
+
+  let totalDrills = 0
+  const drillHits: DrillUsage[] = []
+  for (const [key, count] of Object.entries(state.drillsPerTool)) {
+    totalDrills += count
+    drillHits.push({
+      diameter: state.usedTools[key],
+      count,
+    })
+  }
```

Comment:
If you incremented `totalDrills` at the same time you check `minDrillSize`, this becomes a simple `map`. Same comment applies to `drillRoutes` below

```ts
const drillHits = Object.entries(state.drillsPerTool).map(([key, count]) => ({
  diameter: state.usedTools[key],
  count,
})
```